src/java.base/share/classes/java/lang/String.java

```diff
@@ -845,7 +845,8 @@ private static byte[] encodeWithEncoder(Charset cs, byte coder, byte[] val, bool
         CharsetEncoder ce = cs.newEncoder();
         int len = val.length >> coder;  // assume LATIN1=0/UTF16=1;
         int en = scale(len, ce.maxBytesPerChar());
-        if (ce instanceof ArrayEncoder ae) {
+        // fastpath with ArrayEncoder implies `doReplace`.
+        if (ce instanceof ArrayEncoder ae && doReplace) {
```

Comment:
Wouldn't it be more readable (and perphaps faster) if we do check boolean flag first?

Reply:
Fixed.